### ISSUE TITLE
Ocaml 4.02.2 and Lwt_unix.O_CLOEXEC

### ISSUE THREAD

**Issue by domsj** [ocsigen/lwt#170]
The following fails in utop as of ocaml 4.02.2 (works just fine with 4.02.1)
```
Lwt_unix.openfile
  "/tmp/test"
  [Lwt_unix.O_RDWR; Lwt_unix.O_CLOEXEC; Lwt_unix.O_CREAT; Lwt_unix.O_EXCL] 0o600
 >>= fun fd ->
Lwt_unix.write fd "a" 0 1;;
Exception: Unix.Unix_error (Unix.EBADF, "write", ""). 
```

It not longer fails when I remove the O_CLOEXEC flag.

Doing the same things with (blocking) functions from the standard Unix module does not have this problem...

**Comment by vasilisp** [comment]
I can reproduce this, with or without `utop`. I have no idea what is going on.

**Comment by vbmithr** [comment]
Should be fixed with 2.5.1, please try!

**Comment by rgrinberg** [comment]
:100: @vbmithr thanks for the release
